static/js/components/CompactPostDisplay.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import React from "react"
+import moment from "moment"
+import { connect } from "react-redux"
+import { Link } from "react-router-dom"
+
+import { channelURL, postDetailURL } from "../lib/url"
+import {
+  formatCommentsCount,
+  formatPostTitle,
+  PostVotingButtons
+} from "../lib/posts"
+
+import type { Post } from "../flow/discussionTypes"
+
+class CompactPostDisplay extends React.Component {
+  props: {
+    post: Post,
+    showChannelLink?: boolean,
```

Comment:
It looks like `showChannelLink` is always present, can you make it required here?